plugin/pkg/bastion/validator/admission_test.go

```diff
@@ -0,0 +1,208 @@
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	. "github.com/gardener/gardener/plugin/pkg/bastion/validator"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/testing"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	bastionName = "foo"
+	shootName   = "foo"
+	seedName    = "foo"
+	namespace   = "garden"
+	provider    = "foo-provider"
+	region      = "foo-region"
+)
+
+var _ = Describe("Bastion", func() {
+	Describe("#Admit", func() {
+		var (
+			bastion             *core.Bastion
+			shoot               *core.Shoot
+			coreClient          *corefake.Clientset
+			coreInformerFactory coreinformers.SharedInformerFactory
+			admissionHandler    *Bastion
+		)
+
+		BeforeEach(func() {
+			bastion = &core.Bastion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bastionName,
+					Namespace: namespace,
+				},
+				Spec: core.BastionSpec{
+					ShootRef: corev1.LocalObjectReference{
+						Name: shootName,
+					},
+				},
+			}
+
+			shoot = &core.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      shootName,
+					Namespace: namespace,
+				},
+				Spec: core.ShootSpec{
+					SeedName: pointer.StringPtr(seedName),
+					Provider: core.Provider{
+						Type: provider,
+					},
+					Region: region,
+				},
+			}
+
+			var err error
+			admissionHandler, err = New()
+			Expect(err).ToNot(HaveOccurred())
+			admissionHandler.AssignReadyFunc(func() bool { return true })
+
+			coreClient = &corefake.Clientset{}
+			admissionHandler.SetInternalCoreClientset(coreClient)
+
+			coreInformerFactory = coreinformers.NewSharedInformerFactory(nil, 0)
+			admissionHandler.SetInternalCoreInformerFactory(coreInformerFactory)
+		})
+
+		It("should do nothing if the resource is not a Bastion", func() {
+			attrs := admission.NewAttributesRecord(nil, nil, core.Kind(bastionName).WithVersion("version"), bastion.Namespace, bastion.Name, core.Resource("foos").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+			err := admissionHandler.Admit(context.TODO(), attrs, nil)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should forbid the Bastion creation if a Shoot name is not specified", func() {
+			bastion.Spec.ShootRef.Name = ""
+
+			err := admissionHandler.Admit(context.TODO(), getBastionAttributes(bastion), nil)
+			Expect(err).To(BeInvalidError())
+			Expect(getErrorList(err)).To(ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("spec.shootRef.name"),
+				})),
+			))
+		})
+
+		It("should forbid the Bastion creation if the Shoot does not exist", func() {
```

Comment:
I'm missing a test that allows the Bastion creation

Reply:
I've added one now :)